# HG changeset patch
# User Magnus Melin <mkmelin+mozilla@iki.fi>
# Date 1758807879 -7200
# Node ID 2045d8dc76141000ccc5c8985065d91bebe99ac0
# Parent  72b7126c5ff17afa5a1ffdda017a2babf1d20c7d
Bug 1912195 - Ensure database is commited regularly. r=gds


Commit db of current folder when checking for need to compact, and there was no need.
Also commit after repair folder.

Differential Revision: https://phabricator.services.mozilla.com/D264402

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -3590,8 +3590,9 @@ var folderPane = {
       // In a failure, proceed anyway since we're dealing with problems
       folder.ForceDBClosed();
     }
     folder.updateFolder(top.msgWindow);
+    folder.msgDatabase.commit(Ci.nsMsgDBCommitType.kCompressCommit);
   },
 
   /**
    * Opens the dialog to edit the properties for a folder
diff --git a/mailnews/base/src/nsMsgDBFolder.cpp b/mailnews/base/src/nsMsgDBFolder.cpp
--- a/mailnews/base/src/nsMsgDBFolder.cpp
+++ b/mailnews/base/src/nsMsgDBFolder.cpp
@@ -1696,8 +1696,13 @@ nsresult nsMsgDBFolder::HandleAutoCompac
             rv = AsyncCompactFolders(folderArray, nullptr, aWindow);
           }
         }
       }
+    } else {
+      // Ensure commit happens regularly, even if offline stores are disabled.
+      if (mDatabase) {
+        mDatabase->Commit(nsMsgDBCommitType::kLargeCommit);
+      }
     }
   }
   return rv;
 }
